File: LegacyHealthcareFHIR/LegacyHealthcareFHIR.Core/Mapping/FieldMappingResult.cs
namespace LegacyHealthcareFHIR.Core.Models.Mapping;

public class FieldMappingResult
{
    public int ConfigurationId { get; set; }
    public bool IsApproved { get; set; }
    public List<FieldMappingItemResult> Mappings { get; set; } = new();
}

public class FieldMappingItemResult
{
    public int MappingId { get; set; }
    public required string SourceField { get; set; }
    public required string NormalizedField { get; set; }
    public decimal? AiConfidence { get; set; }
    public string? AiExplanation { get; set; }
}

File: legacy-healthcare-fhir-client/src/app/models/field-mapping-result.ts
export interface FieldMappingResult {
    configurationId: number;
    isApproved: boolean;
    mappings: FieldMappingItemResult[];
}

export interface FieldMappingItemResult {
    mappingId: number;
    sourceField: string;
    normalizedField: string;
    aiConfidence?: number;
    aiExplanation?: string;
}